### PR TITLE
feat: add support for X11 forwarding

### PR DESCRIPTION
**Pull request by kylecarbs**

### CODE REVIEW

File: session_test.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"net"
 	"testing"
 
+	"golang.org/x/crypto/ssh"
```
Comment (deansheather) [inline session_test.go]: .

**Review by deansheather — APPROVED** [review]

File: server.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ type Server struct {
 	PasswordHandler               PasswordHandler               // password authentication handler
 	PublicKeyHandler              PublicKeyHandler              // public key authentication handler
 	PtyCallback                   PtyCallback                   // callback for allowing PTY sessions, allows all if nil
+	X11Callback                   X11Callback                   //callback for allowing X11 forwarding, denies all if nil
```
Comment (mafredri) [inline server.go]:
```suggestion
	X11Callback                   X11Callback                   // callback for allowing X11 forwarding, denies all if nil
```

File: session.go [hunk 3]
```diff
@@ -353,6 +367,23 @@ func (sess *session) handleRequests(reqs <-chan *gossh.Request) {
 				close(sess.winch)
 			}()
 			req.Reply(ok, nil)
+		case "x11-req":
+			if sess.handled || sess.x11 != nil {
```
Comment (mafredri) [inline session.go:371]: Should single connection affect this or is it unrelated to x11-req?

**Review by mafredri — APPROVED** [review]
This is dope!

File: session.go (hunk 3 above)
Comment (kylecarbs) [reply session.go:371]: It should be unrelated. I believe `x11` is the only thing that should respect `SingleConnection`.

**Pull request merged**